_posts/2019-07-16-gentle-intro-to-docker.markdown

```diff
@@ -0,0 +1,65 @@
+
+# Familiar Territory
+
+As a Ruby developer, you've undoubtedly used either RVM or rbenv. We all know why - managing dependencies can be a major pain. Being able to segregate an application into it's own environment with it's own clean slate consisting of only the dependencies that application has drastically simplifies administration and deployment and reduces aggravation. If you're comfortable with this idea, then you're already familiar with the idea of a *container* - a virtual environment used to separate dependencies and configuration from the rest of the system. It not only saves you a lot of hassle, it lets you run different applications with different dependencies on the same machine without any fireworks or panicky phone calls.
+
+# The next logical step
+
+Of course, other developers using other languages and environments long ago figured out that they would like to have the same thing for themselves. In some cases they came up with their own language specific solution to this problem. At the end of the day, though, what you really want is a way to create virtual containers that is language agnostic. Eons ago, when mainframes and workstations roamed the earth, Unix developers came up with the concept of the chroot jail. Although mainly intended for use as a security measure, it inadvertently accomplishes the same sort of dependency isolation we're talking about here, although in a less manageable fashion.  Since it's not specifically meant for deploying applications and managing dependencies, using it in an RVM-ish fashion would be a fairly painful exercise. What would be *really* nice if you could create such environments easily - from a template, for instance - and had a suite of tools to manage them.
+
+# And we widen to reveal....
+
+Enter [Docker](http://docker.com).  It started life as a cloud computing platform, but after a few years, it's creators open sourced it. Open sourcing it did what open sourcing does - changed things - and now we have the modern Docker you've been hearing so much about. But Docker does more than just generalizing the abilities of a language specific manager to any executable application - it brings a number of new abilities to the table. You can migrate a container to an entirely new host, replicate a single container any arbitrary number of times, and encapsulate complex services in an easy to run, atomic fashion. A full listing could (and has) fill a book.  You get the idea.
+
+# Show me
+
+Glad you asked.  The simplest way to get started with Docker is to [install it](https://hub.docker.com/) - you'll have to create an account to get the goodies, and the installation can be a bit of a process, so be ready for that.  Once the smoke has cleared and you're done gritting your teeth, open up a command line and try:
```

Comment:
Docker hub is just an image registry, you don't need to have an account to use docker.

This link might be better to point for docker isntallation https://docs.docker.com/install/